### PR TITLE
Strict-type-checking

### PR DESCRIPTION
**Pull request by DaniFoldi**

### CODE REVIEW

File: src/BetterKV.ts [hunk 1]
```diff
@@ -11,8 +11,8 @@ export declare interface BetterKVGetOptions {
 }
 
 export declare interface BetterKVWithMetadata<V, M> {
-  value: V;
-  metadata: M;
+  value: V | null;
```
Comment (DaniFoldi) [inline src/BetterKV.ts:14]: These two I'm not sure about - the original KV type has them as `V | null`, and in my head nothing is preventing me from calling `.getWithMetadata` where the value is `null`, or the metadata is `null`, but not both (so it's not a simple `null` response.

This feels like a somewhat breaking change, from a typecheck perspective.

File: src/Stubby.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ export class Stubby {
       for(const key of listRet.keys)
         keys.push(this.kv.delete(key.name));
       if(listRet.list_complete) break;
-      cursor = listRet.cursor;
+      if(typeof(listRet.cursor) === 'string') cursor = listRet.cursor;
```
Comment (DaniFoldi) [inline src/Stubby.ts:81]: I have opened an issue in the workers-types repo (https://github.com/cloudflare/workers-types/issues/264), until then this is a good workaround.

File: src/Temra.ts [hunk 3]
```diff
@@ -111,41 +111,45 @@ export class Temra<E = any, O = any> {
         rewriter.on(selector, deleteElement);
         continue;
       }
-      let ElementHandler: HTMLRewriterElementContentHandlers;
+      let ElementHandler: HTMLRewriterElementContentHandlers = {};
```
Comment (DaniFoldi) [inline src/Temra.ts:114]: This one I have no idea about, not tested, but tsc was happy with it.

File: src/Temra.ts [hunk 4]
```diff
@@ -111,41 +111,45 @@ export class Temra<E = any, O = any> {
...
       switch(replacer.type) {
         case "universal":
         case "tagName":
           ElementHandler.element = async (element: Element) => {
+            if (!replacer.replacerFunction) return;
```
Comment (DaniFoldi) [inline src/Temra.ts:119]: These `replacerFunction`s would've thrown if it was called when undefined, so we at least prevent that. Again, not sure if functionally correct.

**Review by helloimalastair — APPROVED** [review]
Looks good, thanks @DaniFoldi!

**Pull request merged**